Source/Core/Core/AchievementManager.cpp

```diff
@@ -55,614 +60,182 @@ void AchievementManager::SetUpdateCallback(UpdateCallback callback)
-      });
-    }
-    if (unlock_status.locked_badge.name != badge_name_to_fetch)
-    {
-      m_image_queue.EmplaceItem([this, index] {
-        std::string current_name, name_to_fetch;
-        {
-          std::lock_guard lock{m_lock};
-          if (m_game_data.num_achievements <= index)
-          {
-            INFO_LOG_FMT(
-                ACHIEVEMENTS,
-                "Attempted to fetch locked badge for index {} after achievement list cleared.",
-                index);
-            return;
-          }
-          const auto& achievement = m_game_data.achievements[index];
-          const auto unlock_itr = m_unlock_map.find(achievement.id);
-          if (unlock_itr == m_unlock_map.end())
-          {
-            ERROR_LOG_FMT(
-                ACHIEVEMENTS,
-                "Attempted to fetch locked badge for achievement id {} not in unlock map.", index);
-            return;
-          }
-          name_to_fetch = achievement.badge_name;
-          current_name = unlock_itr->second.locked_badge.name;
-        }
-        if (current_name == name_to_fetch)
-          return;
-        rc_api_fetch_image_request_t icon_request = {
-            .image_name = name_to_fetch.c_str(), .image_type = RC_IMAGE_TYPE_ACHIEVEMENT_LOCKED};
-        Badge fetched_badge;
-        if (RequestImage(icon_request, &fetched_badge) == ResponseType::SUCCESS)
-        {
-          INFO_LOG_FMT(ACHIEVEMENTS, "Successfully downloaded locked achievement badge id {}.",
-                       name_to_fetch);
-          std::lock_guard lock{m_lock};
-          if (m_game_data.num_achievements <= index)
-          {
-            INFO_LOG_FMT(ACHIEVEMENTS,
-                         "Fetched locked badge for index {} after achievement list cleared.",
-                         index);
-            return;
-          }
-          const auto& achievement = m_game_data.achievements[index];
-          const auto unlock_itr = m_unlock_map.find(achievement.id);
-          if (unlock_itr == m_unlock_map.end())
-          {
-            ERROR_LOG_FMT(ACHIEVEMENTS,
-                          "Fetched locked badge for achievement id {} not in unlock map.", index);
-            return;
-          }
-          if (name_to_fetch != achievement.badge_name)
-          {
-            INFO_LOG_FMT(ACHIEVEMENTS,
-                         "Requested outdated locked achievement badge id {} for achievement id {}.",
-                         name_to_fetch, current_name);
-            return;
-          }
-          unlock_itr->second.locked_badge.badge = std::move(fetched_badge);
-          unlock_itr->second.locked_badge.name = std::move(name_to_fetch);
-        }
-        else
-        {
-          WARN_LOG_FMT(ACHIEVEMENTS, "Failed to download locked achievement badge id {}.",
-                       name_to_fetch);
-        }
-
-        m_update_callback();
-      });
+      u32 achievement_id = achievement_list->buckets[bucket].achievements[achievement]->id;
```

Comment:
Similar to the above, if the achievement access is going to be reused, consider:

```
const auto& achievement = bucket.achievements[achievement];
const u32 achievement_id = achievement->id;
```